cpp/daal/src/algorithms/k_nearest_neighbors/oneapi/bf_knn_classification_predict_kernel_ucapi_impl.i

```diff
@@ -167,47 +187,147 @@ services::Status KNNClassificationPredictKernelUCAPI<algorithmFpType>::compute(c
-                DAAL_CHECK_STATUS_VAR(copyPartialDistancesAndLabels(context, selectResult.values, selectResult.indices, partialDistances,
-                                                                    partialLabels, curQueryRange.count, k, selectionChunkCount,
-                                                                    selectionMaxNumberOfChunks));
-                DAAL_CHECK_STATUS_VAR(labels->releaseBlockOfRows(labelRows));
+                if(computeLabels)
+                {
+                    // Select k smallest distances and their labels from every row of the [curQueryRange.count]x[curDataRange.count] block
+                    DAAL_CHECK_STATUS_VAR(selector->selectNearestDistancesAndLabels(distances, labelRows.getBuffer(), k, curQueryRange.count,
+                                                                                    curDataRange.count, curDataRange.count, 0, selectResult));
+                    DAAL_CHECK_STATUS_VAR(st);
+                    // copy block results to buffer in order to get merged with the same selection algorithm (up to selectionMaxNumberOfChunks of partial results)
+                    // and keep the first part containing previously merged result if exists
+                    DAAL_CHECK_STATUS_VAR(copyPartialDistancesAndLabels(context, selectResult.values, selectResult.indices, partialDistances,
+                                                                        partialLabels, curQueryRange.count, k, selectionChunkCount,
+                                                                        selectionMaxNumberOfChunks));
+                    DAAL_CHECK_STATUS_VAR(labels->releaseBlockOfRows(labelRows));
+                }
+                if(computeRegressionIndices)
+                {
+                    // Select k smallest distances and their indices from every row of the [curQueryRange.count]x[curDataRange.count] block
+                    DAAL_CHECK_STATUS_VAR(selector->selectNearestDistancesAndLabels(distances, blockIndices, k, curQueryRange.count,
+                                                                                    curDataRange.count, curDataRange.count, 0, selectResultIndices));
+                    DAAL_CHECK_STATUS_VAR(st);
+                    // copy block results to buffer in order to get merged with the same selection algorithm (up to selectionMaxNumberOfChunks of partial results)
+                    // and keep the first part containing previously merged result if exists
+                    DAAL_CHECK_STATUS_VAR(copyPartialDistancesAndLabels(context, selectResultIndices.values, selectResultIndices.indices, partialDistances,
+                                                                        partialIndices, curQueryRange.count, k, selectionChunkCount,
+                                                                        selectionMaxNumberOfChunks));
+                }
```

Comment:
The same data is processed twice for combined classification & regression case.

Reply:
Its not the case if only classification is required